src/main/java/com/cosc436002fitzarr/brm/services/SiteAdminUserService.java

```diff
@@ -54,4 +57,56 @@ public String getCreatedSiteAdminMessage() {
         return "CREATED SITE ADMIN USER";
     }
 
+    public User updateSiteAdmin(UpdateUserInput input) {
+        User existingSiteAdmin;
+        try {
+            existingSiteAdmin = userRepository.findByUsername(input.getUsername());
+            LOGGER.info("Successfully retrieved site admin user: " + existingSiteAdmin.toString() + " out of repository to update.");
+        } catch (Exception e) {
+            LOGGER.info(e.toString());
+            throw new RuntimeException(e);
+        }
+        LocalDateTime currentTime = LocalDateTime.ofInstant(Instant.now(), ZoneOffset.UTC);
+
+        EntityTrail updateTrail = new EntityTrail(currentTime, existingSiteAdmin.getId(), getUpdatedSiteAdminMessage());
+
+        List<EntityTrail> existingTrail = existingSiteAdmin.getEntityTrail();
+
+        List<EntityTrail> updatedTrail = new ArrayList<>(existingTrail);
+
+        updatedTrail.add(updateTrail);
+
+        User updatedSiteAdminForPersistence = new User(
```

Comment:
Why not,
`User updatedSiteAdmin = new SiteAdmin(...)` to be more specific here since this a `SiteAdmin` API?